app/assets/stylesheets/components/_header.scss

```diff
@@ -1,21 +1,173 @@
+// Blue box icons in the header, extend as needed. If used elsewhere, abstract from here
+.header-icons {
+  @extend %border-radius;
+  @extend .btn--primary;
+  right: 0;
+  height: 40px;
+  padding: 0;
+  position: relative;
+  float: right;
+  margin-left: $baseline-unit*2;
+  color: #fff;
+  @at-root button.header-icons, & { /* overriding button.btn in _buttons.scss */
+    width: 40px;
+  }
+  .svg-icon {
+    width: 60%;
+  }
+}
+
 .header {
   position: relative;
-  background: #fff;
   z-index: 15;
+  height: 87px;
+  .l-constrained {
+    position: fixed;
+    background: #fff;
+  }
+  @include respond-to($mq-m) {
+    height: auto;
+    .l-constrained {
+      position: relative;
+    }
+  }
 }
 
-.mobile-nav-button {
-  display: block;
-
+.header-content {
+  @extend %clearfix; 
+  box-sizing: border-box;
+  display: flex;
+  padding: $baseline-unit 0;
   @include respond-to($mq-m) {
-    display: none;
+    padding: $baseline-unit*2;
+  }
+}
+
+// Logo
+.header-logo {
+  float: left;
+  @include column(6);
+  
+  .header-logo__link {
+    display: block;
+    height: 75px;
+    @include respond-to($mq-m) {
+      height: 68px;
+    }
+    .header-logo__img {
+      display: block;
+      height: 100%;
+    }
   }
 }
 
-.l-search-bar {
+// interchangable Mobile and desktop content on right
+.header-content_right {
+  @include column(6);
+  float: right;
+  text-align: right;
+}
+.header-content_right-nav {
+  position: relative;
+  top: 50%;
+  transform: translateY(-50%);
+  display: inline-block;
+}
+
+
+// Contact & Get Involved
+.header-links {
+  @extend .header-content_right;
   display: none;
+  @include respond-to($mq-m) {
+    display: inline;
+  }
+  .header-links__link {
+    @extend .header-content_right-nav;
+    a {      
+      padding: $baseline-unit;
+      color: $primary-blue-dark;
+      font-weight: 500;
+    }
+  }
+}
 
+// Mobile Menu & Search button
+.mobile-header__content {
+  @extend .header-content_right;
+  display: block;
   @include respond-to($mq-m) {
-    display: block;
+    display: none;
+  }
+  .mobile-nav-button {
+    @extend .header-content_right-nav;
+    margin: 0;
+    line-height: 1.8rem;
+  }
+  .search__submit--icon {
+    @extend .header-icons;
+    .svg-icon {
+      transform-origin: 0% 0%;
+    }
+    .search-icon__show {
+      transition: transform 300ms ease-out;
+      transform: scale(1) translate(-50%, -50%);
+    }
+    .search-icon__hide {
+      transition: none;
+      transform: scale(0) translate(-50%, -50%);
+    }
   }
 }
+
+// Mobile Search
+.mobile-header__search {
+  position: absolute;
+  overflow: hidden;
+  width: 100%;
+  visibility: hidden;
+  transition: visibility 0ms linear 300ms;
+  .mobile-header__search-bar {
+    display: inline-flex;
+    width: 100%;
+    transform: translateY(-100%);
+    transition: transform 300ms ease-out;
```

Comment:
Maybe add variables for these values (and in the Nav styles) to the _variables.scss file.